app/routes.py

```diff
@@ -1,2 +1,44 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify
 
+class Planet():
+    def __init__(self, id, name, description, rings):
+        self.id = id
+        self.name = name
+        self.description = description
+        self.rings = rings
+
+    def to_json(self):
+        return dict(
+            id = self.id,
+            name = self.name,
+            description = self.description,
+            rings = self.rings)
+
+
+earth = Planet(3, "Earth", "Blue planet", False)
+saturn = Planet(6, "Saturn", "Yellow", True)
+
+planets = [earth, saturn]
+
+bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+@bp.route("", methods=["GET"])
+def handle_planets():
+    planet_list = []
+    for planet in planets:
+        planet_list.append(planet.to_json())
+    return jsonify(planet_list)
+
+@bp.route("/<planet_id>", methods = ["GET"])
+def handle_planet(planet_id):
+    try:
```

Comment:
👀 Be sure to review the Learn info (or review the flasky RT example) for how to refactor the error checking into a helper function. Right now, the main logic of retrieving the details for one record (the route's single responsibility) is obscured by all the error checking. By refactoring, we can both make our error logic reusable, and make the route more readable.